lib/smarter_csv.rb

```diff
@@ -393,15 +393,28 @@ def only_or_except_limit_execution(options, option_name, key)
     def guess_column_separator(filehandle, options)
       skip_lines(filehandle, options)
 
-      possible_delimiters = [',', "\t", ';', ':', '|']
+      delimiters = [',', "\t", ';', ':', '|']
 
-      candidates = if options.fetch(:headers_in_file)
-                     candidated_column_separators_from_headers(filehandle, options, possible_delimiters)
-                   else
-                     candidated_column_separators_from_contents(filehandle, options, possible_delimiters)
-                   end
+      line = nil
+      has_header = options[:headers_in_file]
+      candidates = Hash.new(0)
+      count = has_header ? 1 : 5
+      count.times do
+        line = readline_with_counts(filehandle, options)
+        delimiters.each do |d|
+          candidates[d] += line.scan(d).count
+        end
+      rescue EOFError # short files
+        break
```

Comment:
do you need to rewind before break?  (maybe an `ensure` block?)  is there a test that covers this corner case?